### PR TITLE
[DSY-851] Link

### PR DESCRIPTION
**Pull request by yasminnvaz**
DSY-851 - Component | Link [React Native]

# Description

This adds a new component called Link with two types (standard and underline) following the UI specs and AC described in [DSY-851 - Component | Link [React Native]](https://natura.atlassian.net/browse/DSY-851)

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [X] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# How Has This Been Tested?

To see the changes included, you may want to run the storybook app or sample phone app and check the Link section.

- [X] The Link should be styled as the UI documentation shows

# Checklist:

- [X] My code follows the style guidelines of this project
- [X] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [X] I have made corresponding changes to the documentation
- [X] My changes generate no new warnings
- [X] I have added tests that prove my fix is effective or that my feature works
- [X] New and existing unit tests pass locally with my changes

### CODE REVIEW

File: src/components/Link/Link.stories.tsx [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+  component: Link,
+  title: 'Components/Link',
+};
+
+const onPress = () => {};
+
+export const all = () => (
+  <View style={{ padding: 24 }}>
+    <Link onPress={onPress}>
+      <Text>some content here</Text>
```
Comment (lauralucca) [inline src/components/Link/Link.stories.tsx:15]: Dá pra fazer um parágrafo usando o link no meio? tipo

> ```
> The link component allows you to have a pattern for anchor text. It is intended to be
> used <Link>inside paragraphs</Link> and have the same text styles (size, letter
> spacing, etc) as <Link>the rest of the text content</Link>. 
> ```

Aí pode até usar o underline em um e o sem underline no outro. E se puder escolher o token de tipografia do parágrafo todo seria mais que perfeito!

Comment (yasminnvaz) [reply src/components/Link/Link.stories.tsx]: Feito!

File: src/components/Link/Link.stories.tsx [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+import { TextWithTheme } from '../../common/HelperComponents/ThemeHelper.styles';
+import { Link } from './Link';
+
+export default {
+  component: Link,
+  title: 'Components/Link',
+};
+
+const onPress = () => {
+  Alert.alert('something happend');
```
Comment (lauralucca) [inline src/components/Link/Link.stories.tsx:12]: typo: faltou um `e` em `happened`

Comment (yasminnvaz) [reply src/components/Link/Link.stories.tsx]: opa, corrigido!

File: src/components/Link/Link.stories.tsx [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+};
+
+export const all = () => (
+  <View style={{ padding: 24, width: 336 }}>
+    <TextWithTheme>
+      The link component allows you to have a pattern for anchor text. It is
+      intended to be used <Link onPress={onPress}>inside paragraphs </Link>
+      and have the same text styles (size, letter spacing, etc) as the rest of
+      the{' '}
```
[the remark this answers is not in the document]
Comment (yasminnvaz) [reply src/components/Link/Link.stories.tsx]: boa pergunta, deve ter sido alguma coisinha do atalho de formatação. Acabei de arrumar, obrigada!

**Review by lauralucca — APPROVED** [review]
LGTM

**Review by arielwb — APPROVED** [review]
LGTM

**Pull request merged**